pkg/validation/validate.go

```diff
@@ -123,13 +123,18 @@ func ValidateLabels(limits LabelValidationLimits, tenantID string, ls []*typesv1
 	for _, l := range ls {
 		if len(l.Name) > limits.MaxLabelNameLength(tenantID) {
 			return NewErrorf(LabelNameTooLong, LabelNameTooLongErrorMsg, phlaremodel.LabelPairsString(ls), l.Name)
-		} else if len(l.Value) > limits.MaxLabelValueLength(tenantID) {
+		}
+		if len(l.Value) > limits.MaxLabelValueLength(tenantID) {
 			return NewErrorf(LabelValueTooLong, LabelValueTooLongErrorMsg, phlaremodel.LabelPairsString(ls), l.Value)
-		} else if !model.LabelName(l.Name).IsValid() {
+		}
+		var ok bool
+		if l.Name, ok = SanitizeLabelName(l.Name); !ok {
```

Comment:
The duplicates error message might be a bit confusing, let's say.

With input `hello_world=og` and `hello.world=the-one-with-dots`, you would get something along the lines of:

```
invalid_argument: profile with labels '{..., hello_world="the-one-with-dots", hello_world="og", ...}' has duplicate label name: 'hello_world'
```

Maybe we want to show original `phlaremodel.LabelPairsString(ls)` before modification.